doc/arkode/guide/source/Usage/SPRKStep_c_interface/User_callable.rst

```diff
@@ -0,0 +1,1229 @@
+
+   **Notes:**
+      SPRKStep will not report the initial conditions as a
+      possible zero-crossing (assuming that one or more components
+      :math:`g_i` are zero at the initial time).  However, if it appears
+      that some :math:`g_i` is identically zero at the initial time
+      (i.e., :math:`g_i` is zero at the initial time *and* after the
+      first step), SPRKStep will issue a warning which can be disabled with
+      this optional input function.
+
+
+
+
+
+.. _ARKODE.Usage.SPRKStep.InterpolatedOutput:
+
+Interpolated output function
+--------------------------------
+
+An optional function :c:func:`SPRKStepGetDky()` is available to obtain
+additional values of solution-related quantities.  This function
+should only be called after a successful return from
+:c:func:`SPRKStepEvolve()`, as it provides interpolated values either of
+:math:`y` or of its derivatives (up to the 5th derivative)
+interpolated to any value of :math:`t` in the last internal step taken
+by :c:func:`SPRKStepEvolve()`.
+
+
+
+.. c:function:: int SPRKStepGetDky(void* arkode_mem, realtype t, int k, N_Vector dky)
+
+   Computes the *k*-th derivative of the function
+   :math:`y` at the time *t*,
+   i.e., :math:`y^{(k)}(t)`, for values of the
+   independent variable satisfying :math:`t_n-h_n \le t \le t_n`, with
+   :math:`t_n` as current internal time reached, and :math:`h_n` is
+   the last internal step size successfully used by the solver.  This
+   routine uses an interpolating polynomial of degree *min(degree, 5)*,
+   where *degree* is the argument provided to
+   :c:func:`SPRKStepSetInterpolantDegree()`.  The user may request *k* in the
+   range {0,..., *min(degree, kmax)*} where *kmax* depends on the choice of
+   interpolation module. For Hermite interpolants *kmax = 5* and for Lagrange
+   interpolants *kmax = 3*.
+
+   :param arkode_mem: pointer to the SPRKStep memory block.
+   :param t: the value of the independent variable at which the
+        derivative is to be evaluated.
+   :param k: the derivative order requested.
+   :param dky: output vector (must be allocated by the user).
+
+   :return:
+      * *ARK_SUCCESS* if successful
+      * *ARK_BAD_K* if *k* is not in the range {0,..., *min(degree, kmax)*}.
+      * *ARK_BAD_T* if *t* is not in the interval :math:`[t_n-h_n, t_n]`
+      * *ARK_BAD_DKY* if the *dky* vector was ``NULL``
+      * *ARK_MEM_NULL* if the SPRKStep memory is ``NULL``
+
+   **Notes:**
+      It is only legal to call this function after a successful
+      return from :c:func:`SPRKStepEvolve()`.
+
+      A user may access the values :math:`t_n` and :math:`h_n` via the
+      functions :c:func:`SPRKStepGetCurrentTime()` and
+      :c:func:`SPRKStepGetLastStep()`, respectively.
+
+      Dense outputs may or may not conserve the Hamiltonian. Our testing has
+      shown that Lagrange interpolation typically performs well in this regard,
+      while Hermite interpolation does not. 
```

Comment:
I think this version addresses by comments above about clarifying what is meant by better.